lib/_http_client.js

```diff
@@ -15,6 +15,28 @@ const Agent = require('_http_agent');
 const Buffer = require('buffer').Buffer;
 
 
+// The actual list of disallowed characters in regexp form is more like:
+//    /[^A-Za-z0-9\-._~!$&'()*+,;=/:@]/
+// with an additional rule for ignoring percentage-escaped characters, but
+// that's a) hard to capture in a regular expression that performs well, and
+// b) possibly too restrictive for real-world usage. So instead we restrict the
+// filter to just control characters and spaces.
+//
+// This function is used in the case of small paths, where manual character code
+// checks can greatly outperform the equivalent regexp (tested in V8 5.4).
+function isInvalidPath(s) {
+  var len = s.length;
+  if (len--) { if (s.charCodeAt(0) <= 32) return true; } else return false;
```

Comment:
Is the path not always expected to begin with `/`? If so, the check for s.charCodeAt(0) can look specifically for that.